tcms/search/__init__.py

```diff
@@ -118,6 +110,7 @@ def sum_orm_queries(plans, cases, runs, target):
         if plans:
             cases = cases.filter(plan__in=plans).distinct()
         return cases
+    raise AttributeError('Invalid target')
```

Comment:
The whole module lacks tests. Maybe drop this change for now and do it when tests are added?

Reply:
For this PR you only need a test to verify that the function raises when called with inappropriate parameters. You can add a simple test which calls this function with `objects.none` or `objects.all` queries from the 3 different types and invalid target parameter and assert on the exception.

Reply:
#319

Reply:
Still need the test and to raise `ValueError` instead of `AttributeError`. Everything else on this PR looks fine.

Reply:
Adding this test turned out harder than it looks like, because I can't figure out what arguments does `DjangoSmartQuery` accept. Whatever I do it raises some kind of `Exception` or `.. does not have attribute..`. I will have to deploy and debug it, to see what exactly comes from the request and what is passed to `DjangoSmartQuery`